commit afcc7e5ef289d25c19c7ac881ce505ec910fde7c
Author:     Gregor Beck <gbeck@sernet.de>
AuthorDate: Thu Feb 20 11:25:53 2014 +0100
Commit:     Andreas Schneider <asn@samba.org>
CommitDate: Mon Mar 3 16:49:42 2014 +0100

    s3:winbindd: avoid directly asking a trusted domain in wb_lookupsids*()
    
    As a domain member we should always use a DC of our own domain.
    
    It would be possible to pass all sids in one single dcerpc_wbint_LookupSids()
    call. For now we just fix bug.
    
    Pair-Programmed-With: Stefan Metzmacher <metze@samba.org>
    
    Bug: https://bugzilla.samba.org/show_bug.cgi?id=10458
    Signed-off-by: Gregor Beck <gbeck@sernet.de>
    Signed-off-by: Stefan Metzmacher <metze@samba.org>
    Reviewed-by: Andreas Schneider <asn@samba.org>
    (cherry picked from commit 66fb0ce9557553a4c01607b517e65ac4c93841d0)
---
 source3/winbindd/wb_lookupsids.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source3/winbindd/wb_lookupsids.c b/source3/winbindd/wb_lookupsids.c
index 2fd735d..1bfdba8 100644
--- a/source3/winbindd/wb_lookupsids.c
+++ b/source3/winbindd/wb_lookupsids.c
@@ -320,7 +320,7 @@ static struct wb_lookupsids_domain *wb_lookupsids_get_domain(
 		}
 	}
 
-	wb_domain = find_domain_from_sid_noinit(sid);
+	wb_domain = find_lookup_domain_from_sid(sid);
 	if (wb_domain == NULL) {
 		return NULL;
 	}
